automation_tools/satellite6/upgrade/tests/existence.py

```diff
@@ -0,0 +1,185 @@
+"""Helper functions and variables to test entity existence and associations
+post upgrade
+"""
+
+import os
+import pickle
+from automation_tools.satellite6.upgrade.tasks import csv_reader
+
+# Components for which the post upgrade existence will be validated,
+# non_orged - The components where org is not required to get the data about
+# orged - The components where org is required to get the data about
+components = {
+    'non_orged':
+    [
+        'organization',
+        'host',
+        'capsule',
+        'role',
+        'compute-resource',
+        'user',
+        'subnet',
+        'settings',
+        'os',
+        'hostgroup',
+        'domain'
+    ],
+    'orged':
+    [
+        'content-view',
+        'activation-key'
+    ]
+}
+
+
+def set_datastore(datastore):
+    """
+    Creates a pickle file with all the satellite components data
```

Comment:
Pickle provides protocol parameter to reduce the size of file we will save. But I see more advantages of JSON than pickle. I will manipulate these functions with JSON and update the PR.